### PR TITLE
[Bug] Edit recipe now works with multiple words

### LINKED ISSUE #98: [Enhancement] Edit recipe ingredients

### LINKED ISSUE #174: Edit command should not parse by whitespace
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>
For edit commands, editing of RECIPE_TITLE seems to be restricted to only use of single words. Team may wish to consider having command accept a title that is of 2 words or more, since the original add recipe function does allow RECIPE_TITLE of more than 1 word.

Alternatively, may want to state this restriction on UG.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: WooKaiNing/ped#9

### PR DESCRIPTION
**Pull request by kaijie0102**
fixes #98
fixes #174

### CODE REVIEW

File: src/test/java/essenmakanan/recipe/EditRecipeTest.java [hunk 1]
```diff
@@ -66,4 +68,22 @@ public void editRecipe_invalidInput_exceptionThrown() {
...
+            recipes.editRecipe(recipeToEdit, editDetails);
+        } catch (EssenFormatException e) {
+            e.handleException();
+        }
+
+        String newStep = recipeToEdit.getRecipeSteps().getStepByIndex(0).getDescription();
+        assertEquals("Prepare the dough", newStep);
+        assertEquals("Breads", recipeToEdit.getTitle());
+    }*/
+
```
Comment (StanleyW00) [inline src/test/java/essenmakanan/recipe/EditRecipeTest.java:87]: Is this test going to be used?

File: src/main/java/essenmakanan/command/EditRecipeCommand.java [hunk 2]
```diff
@@ -17,27 +20,122 @@ public EditRecipeCommand(String editDetails, RecipeList recipes) {
...
+
+        for (String attribute : attributesToEdit) {
+            if (!attribute.contains("n/") && !attribute.contains("s/") && !attribute.contains("i/")) {
+                // must have either one of these flags
+                System.out.println("Flag is invalid or format is incorrect.");
+                throw new EssenInvalidEditException();
+            } else if (this.detailsIsNull(attribute)){
+                throw new EssenInvalidEditException();
             }
         }
```
Comment (StanleyW00) [inline src/main/java/essenmakanan/command/EditRecipeCommand.java:122]: Perhaps you should abstract some of the lines. The logic looks good.

**Review by StanleyW00 — CHANGES_REQUESTED** [review]
Overall, LGTM. Just need to make a few changes.

File: src/test/java/essenmakanan/recipe/EditRecipeTest.java (hunk 1 above)
Comment (kaijie0102) [reply src/test/java/essenmakanan/recipe/EditRecipeTest.java:87]: yes, i uncommented it in the next iteration

File: src/main/java/essenmakanan/command/EditRecipeCommand.java (hunk 2 above)
Comment (kaijie0102) [reply src/main/java/essenmakanan/command/EditRecipeCommand.java:122]: will do so after functional codes are done

**Review by StanleyW00 — APPROVED** [review]
Overall, LGTM

**Pull request merged**